gateway/handlers/reader.go

```diff
@@ -11,31 +11,43 @@ import (
 
 	"strings"
 
-	"github.com/openfaas/faas/gateway/metrics"
-	"github.com/openfaas/faas/gateway/requests"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/swarm"
 	"github.com/docker/docker/client"
+	"github.com/openfaas/faas/gateway/metrics"
+	"github.com/openfaas/faas/gateway/requests"
 )
 
 // MakeFunctionReader gives a summary of Function structs with Docker service stats overlaid with Prometheus counters.
 func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
+		verbose := false
+		verboseMode := r.URL.Query().Get("v")
+		if len(verboseMode) != 0 {
+			verbose = true
+		}
 		serviceFilter := filters.NewArgs()
 
 		options := types.ServiceListOptions{
 			Filters: serviceFilter,
```

Comment:
This is not relating to your change, but we may have to optimize this. Can you add a filter here for labels or not? i.e. `function=true`

Reply:
Looks like we can filter for `labels` in both services and tasks.

Services: https://github.com/docker/swarmkit/blob/da5ee2a69176ca3c1eed1b04c4308bc1ab932c05/api/control.pb.go#L301

Tasks: https://github.com/moby/moby/blob/ab0eb8fcf6fe6b4ee12393edcd6465501b5c66a1/daemon/cluster/filters.go#L52

Reply:
That could help us with performance then - as long as the label is propagated to the tasks.